### ISSUE TITLE
Building with --no-default-features fails

### ISSUE THREAD

**Issue by AndrewGaspar** [bsteinb/rsmpi#41]
It looks like too many use statements are being gated on the "user-operations" feature.

It might be worth adding a CI test for this case.

```
pn1707994:rsmpi agaspar$ cargo mpirun --no-default-features --example immediate
   Compiling mpi v0.5.1 (file:///Users/agaspar/Code/rsmpi)
error[E0433]: failed to resolve. Use of undeclared type or module `fmt`
    --> src/collective.rs:1678:6
     |
1678 | impl fmt::Debug for UnsafeUserOperation {
     |      ^^^ Use of undeclared type or module `fmt`

error[E0433]: failed to resolve. Use of undeclared type or module `fmt`
    --> src/collective.rs:1679:27
     |
1679 |     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
     |                           ^^^ Use of undeclared type or module `fmt`

error[E0433]: failed to resolve. Use of undeclared type or module `fmt`
    --> src/collective.rs:1679:46
     |
1679 |     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
     |                                              ^^^ Use of undeclared type or module `fmt`

error[E0412]: cannot find type `c_void` in this scope
    --> src/collective.rs:1705:31
     |
1705 |     unsafe extern "C" fn(*mut c_void, *mut c_void, *mut c_int, *mut ffi::MPI_Datatype);
     |                               ^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
     |
12   | use std::os::raw::c_void;
     |

error[E0412]: cannot find type `c_void` in this scope
    --> src/collective.rs:1705:44
     |
1705 |     unsafe extern "C" fn(*mut c_void, *mut c_void, *mut c_int, *mut ffi::MPI_Datatype);
     |                                            ^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
     |
12   | use std::os::raw::c_void;
     |

error[E0412]: cannot find type `c_int` in this scope
    --> src/collective.rs:1705:57
     |
1705 |     unsafe extern "C" fn(*mut c_void, *mut c_void, *mut c_int, *mut ffi::MPI_Datatype);
     |                                                         ^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
     |
12   | use std::os::raw::c_int;
     |

error: cannot continue compilation due to previous error

error: Could not compile `mpi`.

To learn more, run the command again with --verbose.
```

**Comment by bsteinb** [comment]
Yeah, I broke that when I added the unsafe user operation bits. I'll clean up.

**Comment by bsteinb** [comment]
Should be fixed in 547719184b8b8d54e7249a407bee2a926e1146c1

**Comment by AndrewGaspar** [comment]
Thanks!
